### PR TITLE
cloud: add api overview

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add API overview.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs/pull/9559

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/api-overview.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: TiDB Cloud API Overview
+summary: Learn about what is TiDB Cloud API, its features, and how to use API to manage your TiDB Cloud clusters.
+---
+
+# TiDB Cloud API Overview <span style="color: #fff; background-color: #00bfff; border-radius: 4px; font-size: 0.5em; vertical-align: middle; margin-left: 16px; padding: 0 2px;">Beta</span>
```
Comment (ran-huang) [inline tidb-cloud/api-overview.md:6]: Added `beta` tag as requested by PM 
image

File: tidb-cloud/api-overview.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- **JSON entities.** All entities are expressed in JSON.
+- **HTTPS-only.** You can only access the API via HTTPS, ensuring all the data sent over the network is encrypted with TLS.
+- **Key-based access and digest authentication.** Before you access TiDB Cloud API, you must generate an API key. All requests are authenticated through [HTTP Digest Authentication](https://en.wikipedia.org/wiki/Digest_access_authentication), ensuring the API key is never sent over the network.
+
+To start using TiDB Cloud API, refer to the following resources:
+
+- [Get Started](https://docs.pingcap.com/tidbcloud/api/v1beta#section/Get-Started)
+- [Authentication](https://docs.pingcap.com/tidbcloud/api/v1beta#section/Authentication)
+- [Rate Limiting](https://docs.pingcap.com/tidbcloud/api/v1beta#section/Rate-Limiting)
+- [API Full References](https://docs.pingcap.com/tidbcloud/api/v1beta#tag/Project)
```
Comment (Oreoxmt) [inline tidb-cloud/api-overview.md:35]:
```suggestion
- [API Full References](https://docs.pingcap.com/tidbcloud/api/v1beta#tag/Project)
- [Changelog](https://docs.pingcap.com/tidbcloud/api/v1beta#section/API-Changelog)
```

File: tidb-cloud/api-overview.md (hunk 1 above)
Comment (ran-huang) [reply tidb-cloud/api-overview.md:6]: a preview: https://preview-docs.netlify.app/zh/tidb/dev/api-overview

Comment (qiancai) [reply tidb-cloud/api-overview.md:6]: Will the PDF render this title as expected?

File: TOC-tidb-cloud.md [hunk 3]
```diff
@@ -90,6 +90,7 @@
   - [Scale a TiDB Cluster](/tidb-cloud/scale-tidb-cluster.md)
   - [Upgrade a TiDB Cluster](/tidb-cloud/upgrade-tidb-cluster.md)
   - [Delete a TiDB Cluster](/tidb-cloud/delete-tidb-cluster.md)
+  - [Use TiDB Cloud API (Beta)](/tidb-cloud/api-overview.md)
```
Comment (SunRunAway) [inline TOC-tidb-cloud.md:93]: IMO, it is kindly redundant.
What we need is put API usage in every docs explaining each operation, for example:
image

File: TOC-tidb-cloud.md [hunk 4]
```diff
@@ -400,6 +401,7 @@
         - [`USER_PRIVILEGES`](/information-schema/information-schema-user-privileges.md)
         - [`VIEWS`](/information-schema/information-schema-views.md)
   - [System Variables](/system-variables.md)
+  - [API Reference](https://docs.pingcap.com/tidbcloud/api/v1beta)
```
Comment (SunRunAway) [inline TOC-tidb-cloud.md:404]: We can also put this on our doc home page: https://docs.pingcap.com/tidbcloud/

File: tidb-cloud/api-overview.md (hunk 1 above)
Comment (ran-huang) [reply tidb-cloud/api-overview.md:6]: I'm not sure. I can't generate pdf locally.

File: TOC-tidb-cloud.md (hunk 4 above)
Comment (ran-huang) [reply TOC-tidb-cloud.md:404]: Done in https://github.com/pingcap/docs/pull/10049

File: TOC-tidb-cloud.md (hunk 3 above)
Comment (ran-huang) [reply TOC-tidb-cloud.md:93]: Will do later. But I'm not sure if it's ok to put API usage in every doc when it is still in _beta_.

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Review by SunRunAway — APPROVED** [review]

**Pull request merged**